File: trunk/EngineUAB/Run/Videogame/Data/Scripts/Game/Player/player_animation_idle_state.lua
class 'CPlayerAnimationIdleState' (CState)
	function CPlayerAnimationIdleState:__init() 
		CState.__init(self)
	end

	function CPlayerAnimationIdleState:OnEnter(_CCharacter)
		if not ( _CCharacter == nil ) then
			num = _CCharacter:get_animation_id("idle")
			_CCharacter:get_animation_model():blend_cycle( num, 0.3 )
		end
	end
	
	function CPlayerAnimationIdleState:Execute(_CCharacter)
	end
	
	function CPlayerAnimationIdleState:OnExit(_CCharacter)
		if not ( _CCharacter == nil ) then
			num = _CCharacter:get_animation_id("idle")
			_CCharacter:get_animation_model():clear_cycle( num, 0.3 )
		end
	end
	
	function CPlayerAnimationIdleState:OnMessage(_CCharacter, _Msg)
	end
	
	function CPlayerAnimationIdleState:__Finalize()
	end

File: trunk/EngineUAB/Code/Videogame/Characters/Player/AnimationStates/PlayerAnimationIdleState.cpp
#include "PlayerAnimationIdleState.h"
#include "StatesMachine\Telegram.h"
#include "RenderableObjects\AnimatedModel\AnimatedInstanceModel.h"

#if defined _DEBUG
#include "Memory\MemLeaks.h"
#endif

CPlayerAnimationIdleState::CPlayerAnimationIdleState( const std::string &_Name )
	: CState(_Name)
{

}

CPlayerAnimationIdleState::~CPlayerAnimationIdleState()
{

}

void CPlayerAnimationIdleState::OnEnter( CCharacter* _pCharacter )
{
	int l_iAnimID = _pCharacter->GetAnimationID("idle");
	_pCharacter->GetAnimatedModel()->BlendCycle(l_iAnimID, 1.0f);
}

void CPlayerAnimationIdleState::Execute( CCharacter* _pCharacter, float _fElapsedTime )
{

}

void CPlayerAnimationIdleState::OnExit( CCharacter* _pCharacter )
{
	int l_iAnimID = _pCharacter->GetAnimationID("idle");
	_pCharacter->GetAnimatedModel()->ClearCycle(l_iAnimID, 0.3f);
}

bool CPlayerAnimationIdleState::OnMessage( CCharacter* _pCharacter, const STelegram& _Message )
{
	return false;
}